src/test/java/io/fusionauth/samlv2/service/DefaultSAMLv2ServiceTest.java

```diff
@@ -276,6 +277,33 @@ public void buildLogoutResponse(Binding binding) throws Exception {
     }
   }
 
+  @Test
+  public void buildPostAuthnRequest_signatureFollowsIssuer() throws Exception {
```

Comment:
Should we be testing two scenarios here, one with an element after the `<issuer>` and another w/out? 

This test ends up with a document that looks like this prior to signing.

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<ns3:AuthnRequest xmlns:ns3="urn:oasis:names:tc:SAML:2.0:protocol" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"
                  ID="foobarbaz" IssueInstant="2023-06-29T18:49:20.334Z" Version="2.0">
  <Issuer>https://local.fusionauth.io</Issuer>
  <ns3:NameIDPolicy AllowCreate="false" Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"/>
</ns3:AuthnRequest>
````

Should we also test something like this?

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<ns3:AuthnRequest xmlns:ns3="urn:oasis:names:tc:SAML:2.0:protocol" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"
                  ID="foobarbaz" IssueInstant="2023-06-29T18:49:20.334Z" Version="2.0">
  <Issuer>https://local.fusionauth.io</Issuer>
</ns3:AuthnRequest>
````

Reply:
It is not possible to build an `AuthnRequest` without a `NameIDPolicy` element in the current code, so I don't think we need to test it at this time. (it's also not possible to test it without changing the visibility of some methods in `DefaultSAMLv2Service`)

Reply:
Changing this would require changing the `toAuthnRequest` and similar methods to omit values from the XML if they are `null` on our own domain types (e.g. `AuthenticationRequest`).

Do we want to make that change in logic to be able to test these scenarios at this time?